packages/components/src/date-time/date.js

```diff
@@ -36,7 +36,7 @@ class DatePicker extends Component {
 	}
 
 	render() {
-		const { currentDate } = this.props;
+		const { currentDate, isInvalidDay } = this.props;
```

Comment:
Sorry, I'm nitpicking at this point, but the term "Day" is ambiguous to me, where the value itself to check is a `Date` object. I'm also a bit wary about the negation here, regardless of what `react-dates` is doing, since it seems a bit unnecessary to reverse what is essentially a validity check.

Thoughts on `isValidDate` instead?

There's a further argument about the naming of the prop implying that its value is itself a boolean, when in fact it's a callback function. I'm not sure what precedent there is here for these sorts of callback functions. I'm not overly concerned about it.

Reply:
>I'm also a bit wary about the negation here, regardless of what react-dates is doing, since it seems a bit unnecessary to reverse what is essentially a validity check.

One advantage of the negation relates to the optionality of the prop. By default, all dates are valid. It would seem odd to me that by adding a positively-framed check, `isValidDate`, I'd have to explicitly return true to enable a date that is otherwise enabled by default.

>the term "Day" is ambiguous to me

Great point. 

My slight preference is `isInvalidDate`, but I can go with the positive `isValidDate` as well.

Reply:
I'd be fine with `isInvalidDate`

Reply:
Cool, done and rebased.